hal/mbed_gpio.c

```diff
@@ -66,3 +67,11 @@ void gpio_init_inout(gpio_t *gpio, PinName pin, PinDirection direction, PinMode
         _gpio_init_out(gpio, pin, mode, value);
     }
 }
+
+MBED_WEAK const PinMap *gpio_pinmap()
+{
+    static const PinMap empty_gpio_pinmap[] = {
```

Comment:
Is there a way to populate this for the default case? Maybe for boards with the Arduino form factor this could default to all the Arduino pins.

Reply:
Good idea, I'll add another weak implementation to be used if the Arduino form factor is defined.

Reply:
Done.